atom/renderer/api/lib/desktop-capturer.coffee

```diff
@@ -0,0 +1,22 @@
+ipc = require 'ipc'
+remote = require 'remote'
+NativeImage = require 'native-image'
+
+EventEmitter = require('events').EventEmitter
+desktopCapturer = new EventEmitter
+
+# Tells main process the renderer is requiring 'desktop-capture' module.
+ipc.sendSync 'ATOM_BROWSER_DESKTOP_CAPTURER_REQUIRED'
+
+desktopCapturer.startUpdating = (args) ->
+  ipc.send 'ATOM_BROWSER_DESKTOP_CAPTURER_START_UPDATING', args
+
+desktopCapturer.stopUpdating = () ->
+  ipc.send 'ATOM_BROWSER_DESKTOP_CAPTURER_STOP_UPDATING'
+
+for event_name in ['source-added', 'source-removed', 'source-name-changed', "source-thumbnail-changed"]
```

Comment:
The event name can be sent as a parameter so we don't have to enumerate everyone of them:

```coffeescript
ipc.on 'ELECTRON_RENDERER_DESKTOP_CAPTURE', (event, source) ->
  desktopCapturer.emit event, parseSource(source)
```